### PR TITLE
fix(Core/Creatures): `AttackStart` in `MoveInLineOfSight` is called o…

### LINKED ISSUE #8868: [naxx]Sapphiron P2 bug
### Current Behaviour

from 0d05970e60e0cd4235ea93d1ed19ba5d4bfc1b02 updated to 420ca2e2b57216b90f93e2deb870946b420a72df

`Sapphiron` does not enter P2 now,It was in P1 until death

### Expected Blizzlike Behaviour

`Sapphiron` should enter P2, then throw ice at 3 players in 25 player mode, and will cast an AoE damage enough to kill everyoneThis is a video introduction to the classic world
https://www.youtube.com/watch?v=NwDFC6kFi7c

### Source

_No response_

### Steps to reproduce the problem

1.Fight Sapphiron
2.Wait one minute to see if it enters P2

### Extra Notes

It was always work before 420ca2e2b57216b90f93e2deb870946b420a72df commit

### AC rev. hash/commit

420ca2e2b57216b90f93e2deb870946b420a72df

### Operating system

ubuntu 20.04

### Custom changes or Modules

_No response_

### PR DESCRIPTION
**Pull request by UltraNix**
…nly for aggressive mobs.

Fixes #8868

## Issues Addressed:

- Closes #8868
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/8842

## Tests Performed:

- Not tested.

## How to Test the Changes:

1.Fight Sapphiron
2.Wait one minute to see if it enters P2

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/AI/CreatureAI.cpp [hunk 1]
```diff
@@ -149,7 +149,7 @@ void CreatureAI::MoveInLineOfSight(Unit* who)
                 !me->IsWithinDist(who, ATTACK_DISTANCE, true, false))                      // if in combat and in dist - neutral to all can actually assist other creatures
             return;
 
-    if (me->CanStartAttack(who))
+    if (me->HasReactState(REACT_AGGRESSIVE) && me->CanStartAttack(who))
```
Comment (Nyeriah) [inline src/server/game/AI/CreatureAI.cpp:152]: shouldn't it be?
```
!me->HasReactState(REACT_AGGRESSIVE)
```

**Review by Nyeriah — APPROVED** [review]

**Comment by acidmanifesto** [comment]
Go for this pr. I will working on refactoring the engagement creature ai stuff today and i will work to be aligned with this one so we dont break anything exsisting

**Comment by bgcfb** [comment]
Also #8842

**Comment by Nyeriah** [comment]
> Also #8842

Thanks, I'll merge this as soon as we are able to

**Comment by mpfans** [comment]
work!

**Pull request merged**